examples/suit_update/README.md

```diff
@@ -0,0 +1,556 @@
+    - [Provision the device][setup-wired-provision]
+    - [Configure the network][setup-wired-network]
+  - [Alternative: Setup a wireless device behind a border router][setup-wireless]
+    - [Provision the wireless device][setup-wireless-provision]
+    - [Configure the wireless network][setup-wireless-network]
+  - [Start aiocoap fileserver][start-aiocoap-fileserver]
+- [Perform an update][update]
+  - [Build and publish the firmware update][update-build-publish]
+  - [Notify an update to the device][update-notify]
+- [Detailed explanation][detailed-explanation]
+- [Automatic test][test]
+
+## Prerequisites
+[prerequisites]: #Prerequisites
+
+- Install python dependencies (only Python3.6 and later is supported):
+
+      $ pip3 install --user ed25519 pyasn1 cbor
+
+- Install aiocoap from the source
+
+      $ pip3 install --user --upgrade "git+https://github.com/chrysn/aiocoap#egg=aiocoap[all]"
+
+  See the [aiocoap installation instructions](https://aiocoap.readthedocs.io/en/latest/installation.html)
+  for more details.
+
+- add `~/.local/bin` to PATH
+
+  The aiocoap tools are installed to `~/.local/bin`. Either add
+  "export `PATH=$PATH:~/.local/bin"` to your `~/.profile` and re-login, or execute
+  that command *in every shell you use for this tutorial*.
+
+- Clone this repository:
+
+      $ git clone https://github.com/RIOT-OS/RIOT
+      $ cd RIOT
+
+- In all setup below, `ethos` (EThernet Over Serial) is used to provide an IP
+  link between the host computer and a board.
+
+  Just build `ethos` and `uhcpd` with the following commands:
+
+      $ make -C dist/tools/ethos clean all
+      $ make -C dist/tools/uhcpd clean all
+
+  It is possible to interact with the device over it's serial terminal as usual
+  using `make term`, but that requires an already set up tap interface.
+  See [update] for more information.
+
+## Setup
+[setup]: #Setup
+
+### Key Management
+[key-management]: #Key-management
+
+SUIT keys consist of a private and a public key file, stored in `$(SUIT_KEY_DIR)`.
+Similar to how ssh names its keyfiles, the public key filename equals the
+private key file, but has an extra `.pub` appended.
+
+`SUIT_KEY_DIR` defaults to the `keys/` folder at the top of a RIOT checkout.
+
+If the chosen key doesn't exist, it will be generated automatically.
+That step can be done manually using the `suit/genkey` target.
+
+### Setup a wired device using ethos
+[setup-wired]: #Setup-a-wired-device-using-ethos
+
+#### Provision the device
+[setup-wired-provision]: #Provision-the-device
+
+In order to get a SUIT capable firmware onto the node, run
+
+    $ BOARD=samr21-xpro make -C examples/suit_update clean flash -j4
+
+This command also generates the cryptographic keys (private/public) used to
+sign and verify the manifest and images. See the "Key generation" section in
+[SUIT detailed explanation][detailed-explanation] for details.
+
+From another terminal on the host, add a routable address on the host `riot0`
```

Comment:
Sorry for jumping again late here but I have issues when following the readme of the example application.

At this point the `riot0` interface doesn't exist on my computer so the next command failed. Could we maybe move the next section "Configure the network" before "Provision the device" and remove the `ip address add` command (since setup_network.sh is supposed to do it automatically) ?

Reply:
> At this point the riot0 interface doesn't exist on my computer so the next command failed. Could we maybe move the next section "Configure the network" before "Provision the device" and remove the ip address add command (since setup_network.sh is supposed to do it automatically) ?

This is not done by the script, it it sill needed if you want to use [2001:db8::1], in the test we used [fd00:dead:beef::1] which is configured by the script.

But yest move it, can you push the change right away?

Reply:
In fact, the command below is important but cannot be run before `setup_network.sh` is already running. Another issue that can happen with the `setup_network.sh` script: it cannot be stopped cleanly if the terminal is still opened on the board. In this case, the riot0 is not removed and it's not possible to automatically restart setup_network.sh while the terminal is opened.

Reply:
Yup, makes sense. I moved up "Configure the network" and removed the ip address add command.